src/api/chat.rs

```diff
@@ -349,6 +353,22 @@ async fn get_chat_message_handler(
   Ok(AppResponse::Ok().with_data(messages).into())
 }
 
+#[instrument(level = "debug", skip_all, err)]
+async fn get_chat_question_message_handler(
+  path: web::Path<(String, String)>,
+  query: web::Query<HashMap<String, String>>,
```

Comment:
another approach is to create a struct that derives serde::Deserialize, then you can use web::Query<YourStructName>

Reply:
Thanks for this. Converting result to and from option had me scratching my head